src/main/java/org/wso2/carbon/identity/application/authenticator/social/naver/NaverCustomAuthenticator.java

```
@@ -0,0 +1,433 @@
+					.buildQueryMessage();
+
+			logger.info("authzRequest");
+			logger.info(authzRequest.getLocationUri());
+
+			response.sendRedirect(authzRequest.getLocationUri());
+			logger.info("success");
+		} catch (IOException e) {
+			logger.error("Exception while sending to the login page.", e);
+			throw new AuthenticationFailedException(e.getMessage(), e);
+		} catch (OAuthSystemException e) {
+			logger.error("Exception while building authorization code request.", e);
+			throw new AuthenticationFailedException(e.getMessage(), e);
+		}
+		return;
+	}
+
+	@Override
+	protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+			AuthenticationContext context) throws AuthenticationFailedException {
+
+		logger.info("processAuthenticationResponse");
+		logger.trace("InNaverebookAuthenticator.authenticate()");
+
+		try {
+			if (validateStatusToken(request)) {
+
+				Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+				String clientId = authenticatorProperties.get(NaverCustomAuthenticatorConstants.CLIENT_ID);
+				String clientSecret = authenticatorProperties.get(NaverCustomAuthenticatorConstants.CLIENT_SECRET);
+				String tokenEndPoint = getTokenEndpoint();
+
+				String code = getAuthorizationCode(request);
+				String token = getToken(tokenEndPoint, clientId, clientSecret, code);
+				String nvauthUserInfoUrl = getUserInfoEndpoint();
+
+				logger.debug("code >>>>>>> " + code);
+				logger.debug("token >>>>>>> " + token);
+
+				Map<String, String> requestHeaders = new HashMap<>();
+				requestHeaders.put("Authorization", "Bearer " + token);
+				String responseBody = getUserInfo(nvauthUserInfoUrl, requestHeaders);
+				logger.info(responseBody);
+
+				JSONObject userInfoJson = new JSONObject(responseBody);
+				buildClaims(context, userInfoJson.optJSONObject("response"));
+
+			} else {
+				logger.error("State token validation failed");
```

Shall we throw an exception here?